state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -944,13 +943,33 @@ impl<T: Config<I>, I: 'static> Broadcaster<T::TargetChain> for Pallet<T, I> {
 		(broadcast_id, Self::threshold_sign(api_call, broadcast_id, false))
 	}
 
-	fn threshold_resign(broadcast_id: BroadcastId) -> Option<ThresholdSignatureRequestId> {
-		ThresholdSignatureData::<T, I>::get(broadcast_id)
-			.map(|(api_call, _signature)| Self::threshold_sign(api_call, broadcast_id, false))
+	fn re_sign_broadcast(
+		broadcast_id: BroadcastId,
+		request_broadcast: bool,
+	) -> Result<ThresholdSignatureRequestId, DispatchError> {
+		AbortedBroadcasts::<T, I>::mutate(|aborted| {
+			aborted.remove(&broadcast_id);
+		});
+		let api_call = Self::clean_up_broadcast_storage(broadcast_id)
+			.ok_or(Error::<T, I>::ApiCallUnavailable)?;
+
+		PendingBroadcasts::<T, I>::try_mutate(|pending| {
+			if pending.contains(&broadcast_id) {
+				Err(Error::<T, I>::BroadcastStillPending)
+			} else {
+				if request_broadcast {
+					pending.insert(broadcast_id);
+				}
+				Ok(())
+			}
```

Comment:
Having the broadcast in pending_broadcasts or not will not broadcast or not broadcast it. The ```threshold_sign``` function will cause it broadcast anyways at least once.

Reply:
Not quite sure what you mean here.

`threshold_sign` does not update the PendingBroadcasts, so we need to do it here. 

We also only want to allow broadcasts to be re-signed if they are aborted (otherwise we might end up with the same request having multiple simultaneous broadcast requests). 

Does that make sense?

Reply:
sorry for the confusing language. what i was saying was not correct anyways so we good here.